pandas/_libs/hashtable_class_helper.pxi.in

```diff
@@ -243,17 +243,24 @@ cdef class {{name}}Vector(Vector):
 
     cdef void append(self, {{c_type}} x) noexcept:
 
-        if needs_resize(self.data):
+        if needs_resize(self.data.n, self.data.m):
             if self.external_view_exists:
                 raise ValueError("external reference but "
                                  "Vector.resize() needed")
-            self.resize()
+            self.resize(self.data.m * 4)
```

Comment:
The multiplying by 4 seems somewhat arbitrary and counter-productive for large arrays, but keeping status quo for now